src/plat/bcm2711/overlay-rpi4.dts

```diff
@@ -18,35 +18,60 @@
 		    &{/timer};
 	};
 
+	/*
+	 * See BCM2711 TRM section 1.2 and
+	 * https://github.com/raspberrypi/firmware/issues/1374 for
+	 * reference.
+	 *
+	 * RPi4B seems to be in "low peripheral" mode by default.
+	 * These values were observed from running
+	 * Raspbian instance from /proc/iomem and /sys/firmware/devicetree/base
+	 *
+	 * By default, the RPi 4B has 76MB reserved for the VideoCore/GPU.
+	 * So, we have following memory ranges available:
+	 *
+	 * VC GPU RAM 76MB -> 0x40000000-0x04C00000 = 0x3b400000
+	 *
+	 * 0x00 0x00000000 0x3b400000  948MB
+	 * 0x00 0x40000000 0xbc000000  3008MB
+	 * 0x01 0x00000000 0x80000000  2048MB
+	 * 0x01 0x80000000 0x80000000  2048MB
+	 *
+	 * Disjunct memory areas seem to work only
+	 * when declared in different memory DT nodes.
+	 *
+	 * If all of the declarations below are in the
+	 * same node, the kernel build system seems
+	 * to only understand the second 3008MB area
+	 * and ignores the rest.
+	 */
+
+	/delete-node/ memory;
+
 	memory@0 {
-        /*
-         * See BCM2711 TRM section 1.2 and
-         * https://github.com/raspberrypi/firmware/issues/1374 for
-         * reference.
-         */
+		device_type = "memory";
+		reg = < 0x00000000 0x00000000 0x3b400000 >;
+	};
+
+	memory@40000000 {
+		device_type = "memory";
+		reg = < 0x00000000 0x40000000 0xbc000000 >;
+	};
 
-         /* RPi4B seems to be in "low peripheral" mode by default.
-          * These values were observed from running
-          * Raspbian instance from /proc/iomem and /sys/firmware/devicetree/base
-          *
-          * Set gpu_mem=64 so we have following memory ranges available
-          * VC GPU RAM 64MB -> 0x40000000-0x04000000 = 0x3c000000
-          *
-          * 0x00 0x00000000 0x3c000000  960MB
-          * 0x00 0x40000000 0xbc000000  3008MB
-          * 0x01 0x00000000 0x80000000  2048MB
-          * 0x01 0x80000000 0x80000000  2048MB
-          */
-        reg = < 0x00 0x00000000 0x3c000000 /* 960MB */
-                0x00 0x40000000 0xbc000000 /* 3008MB */
-                0x01 0x00000000 0x80000000 /* 2048MB */
-                0x01 0x80000000 0x80000000 /* 2048MB */ >;
+	memory@100000000 {
+		device_type = "memory";
+		reg = < 0x00000001 0x00000000 0x80000000
+			0x00000001 0x80000000 0x80000000 >;
 	};
 
+	/* TODO: add DMA dedicated ranges
+	 * here also?
+	 */
```

Comment:
What are the dedicated DMA ranges this comment refers to? Can this comment be removed?

Reply:
This comment seems to be a leftover note for myself which can be removed.
I was thinking if all of the reserved memory definitions should be in the same place
for ease of maintaining, as now the DMA pool definition is in `tools/dts/rpi4.dts`:

```
    reserved-memory {
		#address-cells = <0x02>;
		#size-cells = <0x01>;
		ranges;

		linux,cma {
			compatible = "shared-dma-pool";
			size = <0x4000000>;
			reusable;
			linux,cma-default;
			alloc-ranges = <0x00 0x00 0x40000000>;
		};
	};
```

Reply:
Resolved in 979a513.